### ISSUE TITLE
Update .NET Interactive to .NET 9

### ISSUE THREAD

**Issue by jonsequitur** [dotnet/interactive#3760]

**Comment by rbuergi** [comment]
What is the ETA of this?

**Comment by jonsequitur** [comment]
@kimmywork Yes. In the meantime, the workaround is to have the .NET 8 SDK installed as well.

**Comment by Martin-Molinero** [comment]
+1
having net 8 sdk installed doesn't seem to be enough for it to work? net9 builds wont work?
```
error CS1705: Assembly 'XYZ' with identity 'XYZ, Version=2.5.0.0, Culture=neutral, PublicKeyToken=null' uses 
'System.Linq.Expressions, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than 
referenced assembly 'System.Linq.Expressions' with identity 'System.Linq.Expressions, Version=8.0.0.0, Culture=neutral, 
PublicKeyToken=b03f5f7f11d50a3a'
```

```
>dotnet --list-sdks
8.0.403 [C:\Program Files\dotnet\sdk]
9.0.101 [C:\Program Files\dotnet\sdk]
```

```
>dotnet interactive --version
1.0.556801+2d9726b514498cd84403a1d5d13da3161c99eab2
```

**Comment by rbuergi** [comment]
I have mine even compiled for net8.0 and net9.0 and it won't load the dependencies, especially aspnetcore v9, even though they are also built for net8.0 and net9.0... I'm currently not able to use polyglot notebooks at all. Would be really glad to get an update, feeling bad about having upgraded my project to net9, but don't want to go back.

**Comment by laurentpayot** [comment]
It works for me on the latest VS Code on my Ubuntu 24.10 machine with .NET 9 _and_ 8 installed.

**Comment by tmarkovski** [comment]
Does not work with latest `Microsoft.Extensions.AI` which require .net9. Please upgrade!

**Comment by codeape2** [comment]
Will the new version be uploaded to https://www.nuget.org/packages/Microsoft.dotnet-interactive ?

**Comment by jonsequitur** [comment]
This will be available soon. We're working through some issues with the official build pipeline.

**Comment by jonsequitur** [comment]
The .NET 9 version of .NET Interactive is available in the latest pre-release version of the Polyglot Notebooks extension. Please give it a try and let us know if you find any issues.

**Comment by laurentpayot** [comment]
Tried a little bit, no issues found for me 👍

**Comment by tonidy** [comment]
@laurentpayot could you share the steps you use to test .NET Interactive with a pre-release version?

**Comment by laurentpayot** [comment]
@tonidy 
> [@laurentpayot](https://github.com/laurentpayot) could you share the steps you use to test .NET Interactive with a pre-release version?

![Image](https://github.com/user-attachments/assets/e119878c-60a9-41b0-a2f3-df1b6132b74f)

**Comment by jonsequitur** [comment]
> Is there a way to run a c# polyglot notebook on a remote jupyter server with .NET 9?

You can [install the .NET Interactive kernel](https://github.com/dotnet/interactive/blob/main/docs/NotebookswithJupyter.md) like any other Jupyter kernel and connect to it remotely using a Jupyter frontend including Jupyter Lab, VS Code, etc. There are features of the Polyglot Notebooks frontend, though, that won't be available, because they're not supported by Jupyter, e.g. the cell-level kernel chooser, the multi-kernel variable view, etc. 

Or you can use CodeSpaces so that the kernel is running remotely and the Polyglot Notebooks extension works normally.

In either case, the .NET 9 SDK needs to be installed on the server. 

Does that answer the question? Are you looking for something more specific?

**Comment by jonsequitur** [comment]
The .NET 9 version of .NET Interactive is now available with the release version of Polyglot Notebooks.

**Comment by tonidy** [comment]
@jonsequitur I've tried to install the Polyglot Notebooks 1.0.6070010. Here is my extension settings:

<img width="710" alt="Image" src="https://github.com/user-attachments/assets/8cdcfd0d-02f0-43ea-a441-9fb9d22adc45" />

Why the extension needs dotnet interactive 1.0.522904? 

Because I got this error:


```
Process 'dotnet' with PID 95588 exited with code 150 and signal null
Started process 95615: dotnet /Users/user/.nuget/packages/microsoft.dotnet-interactive/1.0.522904/tools/net8.0/any/Microsoft.DotNet.Interactive.App.dll notebook-parser --log-path /Users/user/Projects/dotnet-interactive/parser-logs/
process 95615 stderr: You must install or update .NET to run this application.

App: /Users/user/.nuget/packages/microsoft.dotnet-interactive/1.0.522904/tools/net8.0/any/Microsoft.DotNet.Interactive.App.dll
Architecture: arm64
Framework: 'Microsoft.NETCore.App', version '8.0.0' (arm64)
.NET location: /usr/local/share/dotnet/

The following frameworks were found:
process 95615 stderr: 
  7.0.20 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  9.0.0 at [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=8.0.0&arch=arm64&rid=osx-arm64&os=osx.15
```

<img width="637" alt="Image" src="https://github.com/user-attachments/assets/2e5d3d8f-0856-45a6-bf42-ebac7480fd76" />

<img width="310" alt="Image" src="https://github.com/user-attachments/assets/f97df147-f45a-4855-a102-1d80d1a7ae4c" />

**Comment by jonsequitur** [comment]
@tonidy Did you restart VS Code after updating? There's a bug I just want to rule out -- the `Restart extensions` button isn't enough to trigger the download of the new `dotnet-interactive` tool. If this still happens after a restart, we have a different problem.

**Comment by tonidy** [comment]
@jonsequitur yes, I restarted VS Code after updating and even reinstalled the Polyglot Notebook extension, but the issue still persists.

**Comment by tonidy** [comment]
@jonsequitur found the root cause, while checking this issue https://github.com/dotnet/interactive/issues/3511.  I'm using this setting `dotnet-interactive.kernelTransportArgs` and `dotnet-interactive.notebookParserArgs`, after deleted these settings it's working again. But now there is another problem, I can't connect to another jupyter kernel using `#!connect jupyter`, will create new issue.
